### PR TITLE
#11: Add deploy conditions and credentials to .travis.yml

### PR DESCRIPTION
**Pull request by xander-marjoram**
Publishes to npm on a tagged commit to master.
Using my credentials for now, we can change to fozzie-bot in future.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,5 +12,15 @@ notifications:
...
 
 script:
-  - yarn lint
-  - yarn danger ci
+- yarn lint
+- yarn danger ci
+
+deploy:
+  provider: npm
+  email: xander.marjoram@just-eat.com
```
Comment (DamianMullins) [inline .travis.yml]: is there a way you can change this to ui@just-eat.com in the future so it isn't tied to your email?

Comment (80sgreek) [inline .travis.yml]: Is this going to be an issue if you're *cough* 'not around' in the near future?

File: .travis.yml [hunk 2]
```diff
@@ -12,5 +12,15 @@ notifications:
...
+  api_key:
+    secure: ELvuVI5RvAH4p7xz6VcaM8ibHgiMOect+UK/iE2W96qJwZQXOKM9nIKXXIejclICPdT/noOCvQlNNxet89HynZrv/C4rgxGlcRHdILwf5J2KU1Qyu+Me/S4tpbl70jU5y14RB327j0JsnD9liqst11h7m4qJ7UTsNHZRADWgMVYrwt8aJAkONDS+lhDFilNEB7rWSRGnk66d1AmTAFNxUTiaf78G+xMTQo0zX10HgsWLGuW63DLV9A4Bo+MNXw40j0mSVGctXfBuZ3NvW7+Gnk9HI8jaVZVtfAVHfxQfGr2X+HPTv5/VaPaztiOOT2TswuJhV2Zybe+VbfobL0tG3NigL42ldYao8UaNHrozLxCkLloXZJzbr5vhv/9CiZGd1awJ9+bI8JiRidMwcvsRqrak3YP8e6OPBB/ePcm5bL6TDykyDIhYA7pJTFyTaovbnac3h7wYx6881suj8AY4UYThUcNCqyjiVxfHosH2jNOEsGk53fWj43VfRCXqdjiPUwt75GJjkzxUqvsMTttaX9Z9DNXAIpJ+oXuXbPzZcsOsJMsu6TWwfPx3sf2ILbaZz4IHAob5lB+Gw/CfTJKS40heTzkpkurk+hw/4M6L/8FUlTLNZi22d6KdPfWwxaS6vaqDzW29clovNDKfx/+hIP46wk4ewdUpShu3T0RlVB8=
+  on:
+    tags: true
```
Comment (DamianMullins) [inline .travis.yml]: What will the tag be that it creates? And where does it come from?

Comment (DamianMullins) [reply .travis.yml]: Ah, I think I understand, so will it publish to npm when we do `git push --tags`?

File: .travis.yml (hunk 1 above)
Comment (xander-marjoram) [reply .travis.yml]: @DamianMullins I'm happy to change it now if I can also get a token from that email address

File: .travis.yml [hunk 3]
```diff
@@ -12,5 +12,15 @@ notifications:
...
-  - yarn lint
-  - yarn danger ci
+- yarn lint
+- yarn danger ci
+
+deploy:
+  provider: npm
+  email: xander.marjoram@just-eat.com
+  api_key:
+    secure: ELvuVI5RvAH4p7xz6VcaM8ibHgiMOect+UK/iE2W96qJwZQXOKM9nIKXXIejclICPdT/noOCvQlNNxet89HynZrv/C4rgxGlcRHdILwf5J2KU1Qyu+Me/S4tpbl70jU5y14RB327j0JsnD9liqst11h7m4qJ7UTsNHZRADWgMVYrwt8aJAkONDS+lhDFilNEB7rWSRGnk66d1AmTAFNxUTiaf78G+xMTQo0zX10HgsWLGuW63DLV9A4Bo+MNXw40j0mSVGctXfBuZ3NvW7+Gnk9HI8jaVZVtfAVHfxQfGr2X+HPTv5/VaPaztiOOT2TswuJhV2Zybe+VbfobL0tG3NigL42ldYao8UaNHrozLxCkLloXZJzbr5vhv/9CiZGd1awJ9+bI8JiRidMwcvsRqrak3YP8e6OPBB/ePcm5bL6TDykyDIhYA7pJTFyTaovbnac3h7wYx6881suj8AY4UYThUcNCqyjiVxfHosH2jNOEsGk53fWj43VfRCXqdjiPUwt75GJjkzxUqvsMTttaX9Z9DNXAIpJ+oXuXbPzZcsOsJMsu6TWwfPx3sf2ILbaZz4IHAob5lB+Gw/CfTJKS40heTzkpkurk+hw/4M6L/8FUlTLNZi22d6KdPfWwxaS6vaqDzW29clovNDKfx/+hIP46wk4ewdUpShu3T0RlVB8=
```
Comment (ashleynolan) [inline .travis.yml]: Is it wise to be putting secure API keys into a public repo?

File: .travis.yml (hunk 1 above)
Comment (ashleynolan) [reply .travis.yml]: Is this a token that you get through Travis?

**Comment by kevinrodrigues** [comment]
Not for now, but Is it worth using something like [semantic-release](https://github.com/semantic-release/semantic-release) in the future?

File: .travis.yml (hunk 3 above)
Comment (xander-marjoram) [reply .travis.yml]: This is the encrypted token

**Comment by xander-marjoram** [comment]
Closing this to test on another repo.

**Pull request closed without merging**